### ISSUE TITLE
[ENHANCEMENTS] Allow bypass of Basic Auth for specified subnets

### ISSUE THREAD

**Issue by Sprooty** [tobychui/zoraxy#263]
**Is your feature request related to a problem? Please describe.**
I run services at home that i wish to also access externally, basic auth is a great option for this, however i have seen other similar apps that allow bypass of basic auth for specified subnets, eg 192.168.0/23 or 10.0.0./23

**Describe the solution you'd like**
In basic auth config, allow to specify subnets that will bypass basic auth

**Describe alternatives you've considered**
There is no workaround currently in Zoraxy 

**Additional context**

**Comment by FirefoxNL** [comment]
+1. This could potentially be quite easy to implement with an option of adding an access control list to the basic auth option or, as in NPM, have the option to "satisfy any" so either the access control list OR the basic auth.

**Comment by tobychui** [comment]
@FirefoxNL Thanks for the input. For basic auth, I kinda want to keep it simple (as it is mostly for legacy browsers / embedded machines). 

There is an up-coming Zoraxy SSO + oAuth feature which I guess might be more interesting for these types of uses cases. Though, I will still keep this here just in case someone want to pick it up and create a PR 👍🏻

**Comment by neumeier-cloud** [comment]
+1, this feature would be great just like npm auth if not whitelisted + access ip
![image](https://github.com/user-attachments/assets/da56827a-f75e-46cd-abb7-55080658882d)
![image](https://github.com/user-attachments/assets/51cb6d27-e522-4d08-8cea-a10650880dd7)
![image](https://github.com/user-attachments/assets/a0e6adab-350f-4390-96a3-1b05b12665c6)

**Comment by deluxestyle** [comment]
I would like this as well. This is the only feature that keeps me away from zoraxy

**Comment by king8084** [comment]
not joking, i went back to npm just because of this, zoraxy >>>>>> npm, but the fact i cant auth some apps unless i can bypass by making them "Trusted ips" is an absolute deal breaker for me

**Comment by DarinDev1000** [comment]
I would also like this feature, to be able to add basic username/password auth on a proxy

**Comment by tobychui** [comment]
Finally got time to implement this after a year or so. 

<img width="829" height="631" alt="Image" src="https://github.com/user-attachments/assets/45c33dc5-9e93-4919-8c10-a6f577895592" />

In the next release you will be able to add IP / CIDR into exclusion list for basic auth.

**Comment by FirefoxNL** [comment]
@tobychui That looks very good! Looks like this is now implemented per proxy, which is nice! Have you considered implementing this on the access control? I feel like that already covers security and allows you to configure it once and use the exclusion for multiple hosts. :)

**Comment by tobychui** [comment]
@FirefoxNL Nope, as Authentication Provider and Access Control Rules are two separated modules, they cant be merged together easily without some big rewrite.

**Comment by AnthonyMichaelTDM** [comment]
can't IP addresses be spoofed?

**Comment by tobychui** [comment]
Closing this as the feature has already been implemented. But in general you should use path prefix instead of IPs as mentioned above, IP can be spoofed and it is not possible for Zoraxy to get client IP in 100% accuracy.
